Avoid using variable-length array

ericlinsechs opened the pull request:
Using variable-length array is not a good call
([https://lkml.org/lkml/2011/10/23/25](https://lkml.org/lkml/2011/10/23/25)). So remove it.

jserv reviewed (changes requested):
Check https://cbea.ms/git-commit/ carefully and refine the git commit message.

Hyperlinks should appear at the end of git commit message.

ericlinsechs commented:
I have corrected the commit message. Could you please review it? Thank you.

jserv commented:
> I have corrected the commit message. Could you please review it? Thank you.

You don't have to leave the above message. Simply click "Re-request review" via GitHub web page.
See https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/proposing-changes-to-your-work-with-pull-requests/requesting-a-pull-request-review

jserv reviewed (changes requested):
Don't place "Link: https://lkml.org/lkml/2011/10/23/25", which does not express the rationale. You can quote some insights from the LKML.

jserv commented on client.c:
```diff
@@ -29,6 +29,10 @@ int main()
     for (int i = 0; i <= offset; i++) {
         lseek(fd, i, SEEK_SET);
         sz = read(fd, buf, 1);
+        if (sz < 0) {
```
This should appear in another pull request. Let's stick to the proposed subject for necessary changes.

jserv commented on client.c:
```diff
@@ -38,6 +42,10 @@ int main()
     for (int i = offset; i >= 0; i--) {
         lseek(fd, i, SEEK_SET);
         sz = read(fd, buf, 1);
+        if (sz < 0) {
```
Ditto. Decouple from this commit.

jserv commented on fibdrv.c:
```diff
@@ -27,8 +28,9 @@ static int major = 0, minor = 0;
-    long long f[k + 2];
+    long long *f = kmalloc(sizeof(*f) * (k + 2), GFP_KERNEL);
+    if (!f)
+        return -1;
```
The return value `-1` should be documented, otherwise it is nonsensical as for the math definition.

ericlinsechs replied:
I supposed `errno` would be appropriate to describe the situation?

`   return -ENOMEM;`

jserv commented on fibdrv.c:
```diff
@@ -27,8 +28,9 @@ static int major = 0, minor = 0;
-    long long f[k + 2];
+    long long *f = kmalloc(sizeof(*f) * (k + 2), GFP_KERNEL);
+    if (!f)
+        return -ENOMEM;
```
Provide explicit comments for this function.

ericlinsechs committed:
Avoid using variable-length array

The size is always valid, but variable-length arrays generate worse code
for no good reason, unless the function happens to be inlined and the
compiler sees the length for the simple constant it is.

Signed-off-by: Eric Lin <ericlinsechs@gmail.com>

jserv commented:
Thank @ericlinsechs for contributing!
